### PR TITLE
Feature: block ordering

### PR DESCRIPTION
**Pull request by neotyk**
Introduces `common-db/orderkeeper`

### CODE REVIEW

**Review by filipesilva — APPROVED** [review]

**Comment by filipesilva** [comment]
Awesome work!

Carve still unhappy tho :p

File: src/clj/athens/self_hosted/web/datascript.clj [hunk 1]
```diff
@@ -65,13 +67,28 @@
...
+          ;; normalize txs
+          txs     (if (map? txs)
+                    [txs]
+                    txs)]
+      (log/debug "event-id:" event-id ", normalized-txs:" (with-out-str
+                                                            (pprint/pprint txs)))
+      (let [processed-tx            (->> txs
+                                         (common-db/linkmaker @connection)
+                                         (common-db/orderkeeper @connection))
+            _                       (log/debug event-id "after tx-middleware:" (with-out-str
```
[the remark this answers is not in the document]
Comment (neotyk) [reply src/clj/athens/self_hosted/web/datascript.clj:84]: I agree

**Comment by neotyk** [comment]
> Awesome work!
> 
> 
> 
> Carve still unhappy tho :p

Carve is never happy ;-)

File: src/clj/athens/self_hosted/web/datascript.clj (hunk 1 above)
[the remark this answers is not in the document]
Comment (neotyk) [reply src/clj/athens/self_hosted/web/datascript.clj]: Removed it.
If we need to debug it, we put it there by hand.

**Pull request merged**